<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Inside of a interpretable-by-desing ML model</title>
  <link rel="stylesheet" href="styles/chart_styles.css">
  <link rel="stylesheet" href="styles/psi_chart_styles.css">
  <link rel="stylesheet" href="styles/footer.css">
  <link rel="stylesheet" href="styles/styles.css">
  <link rel="stylesheet" href="styles/psi_video_styles.css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css">
  <link rel="stylesheet" href="styles/networkVideos.css">

  <link href="https://fonts.googleapis.com/css2?family=Merriweather:wght@400;700;900&display=swap" rel="stylesheet">

  <script type="text/javascript" src="https://d3js.org/d3.v4.js"></script>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/d3/7.8.5/d3.min.js"></script>
</head>

<body>
  <script src="functions.js"></script>
  <script src="constants.js"></script>
  <script src="utils.js"></script>
  <script src="psiChart_nonLinear.js"></script>
  <script src="comparison.js"></script>


  <header>

    <header class="full-width-header">
      <div class="header-content">
        <h1>Inside an interpretable-by-design machine learning model: enabling RNA splicing rational design</h1>
        <!-- <p class="subtitle">Subtitle</p> -->
        <div class="author-info">
          <p><span class="author">Mateus Aragao</span>, <span class="author">Shiwen Zhu</span>, <span class="author">Susan Liao</span>, <span class="author">Nhi Nguyen</span></p>

          <p class="affiliation">Department of Computer Science, New York University</p>
          <!-- <p>Published on <time datetime="2024-07-09">July 9, 2024</time></p> -->
        </div>
      </div>
    </header>

    <main class="content-wrapper">
      <article>

        <section>
          <h2>Abstract</h2>
          <p>Deciphering the regulatory logic of RNA splicing, a critical process in genome function, 
            remains a major challenge in modern biology. While various machine learning models have 
            been proposed to address this issue, many of them fall short in terms of interpretability, 
            unable to articulate how they arrive at their predictions. We recently introduced an 
            interpretable machine learning model that predicts splicing outcomes based on input 
            sequence and structure. Here, we present a series of interactive data visualization tools 
            to illuminate the process behind the network's predictions. Specifically, we introduce 
            visualizations that emphasize both the global and local interpretability of our model. 
            These visualizations emphasize the clear intermediate reasoning stages of our model that 
            trace how specific RNA features contribute to the final splicing prediction. We highlight 
            how these visualizations can be used to explain the network’s performance on prior 
            training and validation datasets. Finally, we explore how these interactive visualizations 
            can be harnessed to facilitate domain-specific applications, such as rational design of 
            RNA sequences with desired splicing outcomes. Together, these visualizations highlight 
            the role of data visualization and interactivity in enhancing machine learning 
            interpretability and model adoption.</p>
        </section>
        
        <h2>Background</h2>

        <h3><b>RNA Splicing</b></h3>
        <p>Genetic information stored in the human genome in DNA is used as a template for RNA. <br><br>
          For many genes in the human genome, different RNA products can be produced from the same DNA. <br><br>
          The process of generating these different RNA products is termed alternative RNA splicing.<br><br>
          The most common form of alternative splicing involves the joining of differential RNA exon sequences.
        </p>

        <h3><b>Exon Inclusion</b></h3>
        <p>
          In the case of exon inclusion, an exon is included in the spliced product.
        </p>
        <div class="container">
          <img id="previewImage" src="resources/RNA_Splicing.001.jpeg" alt="Preview Image">
          <i class="fas fa-arrow-right"></i>
          <video id="myVideo" autoplay muted>
            <source src="resources/RNA_Splicing_Inclu.m4v" type="video/mp4">
            Your browser does not support the video tag.
          </video>
          <button id="replayButton"><i class="fas fa-redo"></i></button>
        </div>

        <h3><b>Exon Skipping</b></h3>
        <p>
          In the case of exon skipping, an exon is removed from the spliced product.
        </p>
        <div class="container">
          <img id="previewImage" src="resources/RNA_Splicing.001.jpeg" alt="Preview Image">
          <i class="fas fa-arrow-right"></i>
          <video id="myVideo2" width="480" height="270" autoplay muted>
            <source src="resources/RNA_Splicing_Skip.m4v" type="video/mp4">
            Your browser does not support the video tag.
          </video>
          <button id="replayButton2"><i class="fas fa-redo"></i></button>
        </div>

        <h3><b>Percent Spliced In (PSI)</b></h3>
        <p>
          When information is read out from the genome (transcribed), multiple RNAs are produced for each gene.
          These RNAs are then alternatively spliced, often generating both inclusion and skipping products.
          For example, a hundred RNAs can be transcribed from the same DNA sequence.
        </p>
        <!-- PSI Video Section -->
        <div class="video-section-container">
          <section id="video-section">
            <h3>Input (Exon Sequence and Exon Structure)</h3>
            <div class="video-buttons">
              <button class="styled-button one" onclick="selectVideo('video1', 'info1', 'one')">Exon 1</button>
              <button class="styled-button two" onclick="selectVideo('video2', 'info2', 'two')">Exon 2</button>
              <button class="styled-button three" onclick="selectVideo('video3', 'info3', 'three')">Exon 3</button>
            </div>
            <div id="info-container" class="info-container">
              <img src="" alt="Info Image" id="info-image">
            </div>
            <h3>Output (Result and PSI)</h3>
            <div class="video-container">
              <video id="video1" autoplay muted>
                <source src="resources/psiVideos/PSI_New_1.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
              <video id="video2" autoplay muted>
                <source src="resources/psiVideos/PSI_New_2.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
              <video id="video3" autoplay muted>
                <source src="resources/psiVideos/PSI_New_3.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
            </div>
          </section>
        </div>
        <script src="psiVideos.js"></script>

        <p>Structure explanation (animation)
        </p>
        <div class="container">
          <video id="myVideo3" autoplay muted>
            <source src="resources/networkVideos/Secondary_structure.m4v" type="video/mp4">
            Your browser does not support the video tag.
          </video>
          <button id="replayButton3"><i class="fas fa-redo"></i></button>
        </div>

        <h3><b>Splicing Logic</b></h3>
        <p>
          One of the major challenges in modern genomics research is understanding the logic underlying RNA splicing,
          specifically understanding which RNA-encoded features lead to a specific PSI.
          While RNAs encode two-dimensional information (sequence),
          they can also form more complex three-dimensional structures (structures).
          <br><br>
          <br><br>
          <br><br>
        </p>

        <!-- <div class="chart-container">
          <div class="psi-chart-gradient" id="psi-chart-gradient"></div>
        </div> -->

        <h3>Input</h3>
        <p>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique
          cursus.
          Nam
          nulla quam, gravida non, commodo a, sodales sit amet, nisi.</p>
        <h3>Output</h3>
        <p>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique
          cursus.
          Nam
          nulla quam, gravida non, commodo a, sodales sit amet, nisi.</p>
        <p>3.PSI definition (interactive?)</p>

        <hr>
        <h2>Interpretable-by-design network</h2>
        <!-- Network Video Section -->
        <div class="network-video-section-container" id="network-section-1">
          <section id="network-video-section-1">
            <div class="network-buttons">
              <div class="selector"></div>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-1', 'networkVideo1', 0)">Step 1</button>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-1', 'networkVideo2', 1)">Step 2</button>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-1', 'networkVideo3', 2)">Step 3</button>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-1', 'networkVideo4', 3)">Step 4</button>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-1', 'networkVideo5', 4)">Step 5</button>
            </div>
            <div class="network-video-container">
              <video id="networkVideo1" autoplay muted preload="auto">
                <source src="resources/networkVideos/Network_1.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
              <video id="networkVideo2" autoplay muted preload="auto">
                <source src="resources/networkVideos/Network_2.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
              <video id="networkVideo3" autoplay muted preload="auto">
                <source src="resources/networkVideos/Network_3.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
              <video id="networkVideo4" autoplay muted preload="auto">
                <source src="resources/networkVideos/Network_4.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
              <video id="networkVideo5" autoplay muted preload="auto">
                <source src="resources/networkVideos/Network_5.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
            </div>
          </section>
        </div>
        <script src="networkVideos.js"></script>

        <h3>Learned features </h3>
        <p>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique
          cursus.
          Nam
          nulla quam, gravida non, commodo a, sodales sit amet, nisi.</p>

        <p>6.Non-linear relationship between a.u. and PSI (interactive)
          Click along line to generate difference to prediction plot
        </p>

        <!-- <div class="psi-view" id="psi-view">
          <svg class="psi-view" preserveAspectRatio="xMidYMid meet"></svg>
        </div> -->
        <div class="chart-container">
          <div id="non-linear-chart" class="non-linear-chart"></div>
          <div id="psi-bar-chart" class="psi-bar-chart"></div>
        </div>
        <h3>7./8.Feature visualization (interactive)</h3>

        <div class="feature-nucleotide">
          <!-- <button id="switch-dataset-button">Switch Dataset</button> -->

          <div class="nucleotide-view" id="nucleotide-view">
            <svg class="nucleotide-view" width="100%" height="100%"></svg>
          </div>
          <div class="bottom">
            <div class="feature-legend-container">
              <div class="feature-legend-grids">
                <div class="feature-legend-title">Feature Legend</div>
                <div class="legend"></div>
                <div class="feature-grids-wrapper">
                  <div class="svg-grid-inclusion"></div>
                  <div class="svg-grid-skipping"></div>
                </div>
                <div class="svg-grid-long-skipping"></div>
              </div>
            </div>
            <div class="nucleotide-mini-charts">
              <div class="nucleotide-sort">
                <svg class="nucleotide-sort" width="100%" height="100%"></svg>
              </div>
              <div class="nucleotide-zoom">
                <svg class="nucleotide-zoom" width="100%" height="100%"></svg>
              </div>
            </div>
          </div>
        </div>

        <p>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique
          cursus.
          Nam
          nulla quam, gravida non, commodo a, sodales sit amet, nisi.</p>

        <h3>Class strengths</h3>
        <p>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique
          cursus.
          Nam
          nulla quam, gravida non, commodo a, sodales sit amet, nisi.</p>
        <div class="feature-view-1">
          <svg class="feature-view-1" width="100%" height="100%"></svg>
        </div>

        <h3>Difference to prediction</h3>
        <p>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique
          cursus.
          Nam
          nulla quam, gravida non, commodo a, sodales sit amet, nisi.</p>

        <!-- Local Network Video Section -->
        <div class="network-video-section-container" id="network-section-2">
          <section id="network-video-section-2">
            <div class="network-buttons">
              <div class="selector"></div>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo1', 0)">Step 1</button>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo2', 1)">Step 2</button>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo3', 2)">Step 3</button>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo4', 3)">Step 4</button>
              <button class="network-styled-button"
                onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo5', 4)">Step 5</button>
            </div>
            <div class="network-video-container">
              <video id="newNetworkVideo1" autoplay muted preload="auto">
                <source src="resources/networkVideos/Network_1.m4v" type="video/mp4"> Your browser does not support the video tag.
              </video>
              <video id="newNetworkVideo2" autoplay muted preload="auto">
                <source src="resources/networkVideos/Local_Network_Feature.m4v" type="video/mp4"> Your browser does not support the video tag.
              </video>
              <video id="newNetworkVideo3" autoplay muted preload="auto">
                <source src="resources/networkVideos/Local_Network_Add.m4v" type="video/mp4"> Your browser does not support the video tag.
              </video>
              <video id="newNetworkVideo4" autoplay muted preload="auto">
                <source src="resources/networkVideos/Local_Network_Subt.m4v" type="video/mp4"> Your browser does not support the video tag.
              </video>
              <video id="newNetworkVideo5" autoplay muted preload="auto">
                <source src="resources/networkVideos/Local_Network_PSI.m4v" type="video/mp4"> Your browser does not support the video tag.

              </video>
            </div>
          </section>
        </div>

        <hr>

        <h2>Applications</h2>


        <h3>Mutations</h3>

        <p>A common task to probe interpretability is to examine how the network predicts an outcome when an input is
          edited.
          Tracing the consequences of input editing is often highly domain-specific; for example, natural language
          processing
          researchers examine how altering a single character can affect meaning (e.g. adding ``s'' to the word ``cat'')
          and
          vision researchers examine how altering a set of pixels can affect classification. Thus, a common experimental
          approach
          in molecular biology is to perform single nucleotide mutagenesis - substituting one nucleotide for another
          (e.g. replacing
          A with G at a specific position in an exon). For machine learning researchers, mutagenesis provides an
          approach to probe whether
          the local interpretability of a neural network generalizes beyond the training and testing datasets
          \cite{whatiftool}. For domain
          experts in molecular biology, whether \textit{in silico} mutagenesis predictions match results from laboratory
          mutagenesis experiments
          often serves as a litmus test for whether a neural network captures the ground truth. Furthermore, mutagenesis
          experiments are often used to
          model real-world disease mutations as single nucleotide splicing mutations have been directly implicated in
          several diseases \cite{lorson1999single,ezquerra2017high,ars2000mutations, teraoka1999splicing},
          and are estimated to contribute to up to 30\% of rare genetic disorders \cite{lim2011using}.
        </p>
        <p>
          To illustrate how our visualizations can be useful for mutagenesis modeling,
          we generate exon view visualizations for a single mutation from Liao, Sudarshan,
          and Regev's original paper (\cref{fig:mutation}). Each visualization elucidates
          important facets of the network's local reasoning process. Comparing class strengths
          charts between the original input and the single mutation reveals that the single mutation
          shifted the dominant feature class from skipping to inclusion (\cref{fig:mutation}, left panels).
          Examining the feature-focused charts reveals that the mutation resulted in a two-fold decrease in a
          skipping feature (\cref{fig:mutation}, middle panels). Finally, comparing exon visualizations
          illustrates the consequences of the single nucleotide mutation at position 34. Examining the RNAfold
          on \textit{both} inclusion and skipping features at position 34.

          We further demonstrate that our visualizations can also extend to a double mutation scenario
          (Supplemental Fig.~S5).
        </p>
        <p>
          As reflected in the class strengths chart, the additional mutation at position 46 results in an
          increase in skipping class strengths. Examining the feature-focused chart reveals that the second
          mutation increased the strength of the secondary structure feature. Analysis of structure in exon
          view reveals that the second mutation at position 46 re-establishes base pairing with the first mutation
          at position 34, restoring the strong skipping feature.
        </p>

        <p>
          By comparing visualizations across the original input, single mutation, and double mutation, scenarios,
          we illuminate the local reasoning processes of the neural network. These visualizations not only provide
          a means to assess the generalization of interpretability beyond training and testing datasets but also
          serve as a bridge between \textit{in silico} predictions and real-world mutagenesis experiments.
          This alignment with real-world molecular biology scenarios, such as disease mutations implicated in
          genetic disorders, underscores the utility and relevance of our visualizations in enhancing the
          interpretability
          and applicability of this neural network model in advancing molecular biology research.</p>

        <!-- <button id="s1-exon">S1 Exon</button>
        <button id="s1-mutation">S1 Mutation</button> -->
        <!-- <div class="chart-container">
          <svg id="nucleotide-view-exon-s1" class="nucleotide-view-exon-s1"></svg>
      </div> -->
        <div class="chart-container">
          <svg id="nucleotide-view-exon-s1-mutation" class="nucleotide-view-exon-s1-mutation"></svg>
        </div>

        <h3>Comparisons</h3>
        <p>One key aspect of interpretability is comparing the network’s reasoning
          process across multiple inputs. At the local level, this involves examin-
          ing how the network’s complex reasoning differs across input instances,
          specifically examining how the network identifies different features
          across inputs yet follows the same reasoning process to arrive at similar
          predictions. By analyzing reasoning across multiple input instances,
          researchers can also gain insights into the global reasoning processing
          within a network. </p>
        <p>
          Our first use case demonstrates how these visualizations can be used
          to examine how the network can arrive at the same output prediction
          from three different inputs from the original dataset. We
          generated difference-to-prediction charts, strongest skipping feature
          charts, and exon view visualizations for all three inputs. While
          the difference-to-prediction view is the same across the two inputs and
          the feature class view appears very similar, closer inspection reveals that
          the specific inclusion and skipping values vary for each input. Drilling
          down into the hierarchical feature-focused charts for each sequence also
          reveals different specific skipping features. Furthermore, a comparison
          of exon views reveals differences in the distribution of feature strengths
          across the three inputs.
        </p>
          <div class="chart-container">
              <svg id="nucleotide-view-exon1" class="nucleotide-view-exon1"></svg>
              <div id="psi-bar-chart-s1" class="psi-bar-chart-s1"></div>
          </div>
          <br>
          <div class="chart-container">
              <svg id="nucleotide-view-exon-comp" class="nucleotide-view-exon-comp"></svg>
              <div id="psi-bar-chart-s1-comparison" class="psi-bar-chart-s1-comparison"></div>
          </div>

          <!-- todo  -->
          <div class="chart-container">
            <svg id="nucleotide-view-exon-brca2" class="nucleotide-view-exon-brca2"></svg>
            <div id="psi-bar-chart-exon-brca2" class="psi-bar-chart-exon-brca2"></div>
        </div>

        <div class="chart-container">
          <svg id="nucleotide-view-exon-cfrt" class="nucleotide-view-exon-cfrt"></svg>
          <div id="psi-bar-chart-exon-cfrt" class="psi-bar-chart-exon-cfrt"></div>
      </div>

      <h3>3x3 table with same psi</h3>

<!-- 3x3 grid of graphs -->
<div class="chart-container">
  <div class="grid-container">
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-1" class="nucleotide-view-exon-grid"></svg>
    </div>
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-2" class="nucleotide-view-exon-grid"></svg>
    </div>
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-3" class="nucleotide-view-exon-grid"></svg>
    </div>
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-4" class="nucleotide-view-exon-grid"></svg>
    </div>
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-5" class="nucleotide-view-exon-grid"></svg>
    </div>
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-6" class="nucleotide-view-exon-grid"></svg>
    </div>
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-7" class="nucleotide-view-exon-grid"></svg>
    </div>
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-8" class="nucleotide-view-exon-grid"></svg>
    </div>
    <div class="grid-item">
      <svg id="nucleotide-view-exon-s1-9" class="nucleotide-view-exon-grid"></svg>
    </div>
  </div>
</div>

        <h2>Summary</h2>
        <p>Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum
          tortor
          quam, feugiat vitae, ultricies eget, tempor sit amet, ante.</p>
        <div class="collapsible-section">
          <button class="collapsible button">Learn More(example of a foldable section)</button>
          <div class="content">
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et
              dolore
              magna aliqua.</p>
          </div>
        </div>

        <!-- Rest of your content goes here -->
      </article>
    </main>
    <footer class="article-footer">
      <div class="footer-content">
        <section>
          <h3>Acknowledgment</h3>
          <p>Acknowledgments here</p>
        </section>

        <section>
          <h3>Discussion and Review</h3>
          <p>if there are any discussions</p>
        </section>

        <section>
          <h3>Further Reading</h3>
          <ul>
            <li><a href="#">Link to interface</a></li>
            <li><a href="#">Link to paper?</a></li>
          </ul>
        </section>

        <section>
          <h3>References</h3>
          <ol>
            <li>formatted references</li>
          </ol>
        </section>

        <section>
          <h3>Appendix</h3>
          <p>additional resources</p>
        </section>
      </div>
    </footer>
</body>

</html>