<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Inside an interpretable-by-design machine learning model: enabling RNA splicing rational design</title>
  <link rel="stylesheet" href="styles/chart_styles.css">
  <link rel="stylesheet" href="styles/psi_chart_styles.css">
  <link rel="stylesheet" href="styles/footer.css">
  <link rel="stylesheet" href="styles/styles.css">
  <link rel="stylesheet" href="styles/psi_video_styles.css">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css">
  <link rel="stylesheet" href="styles/networkVideos.css">

  <link href="https://fonts.googleapis.com/css2?family=Merriweather:wght@400;700;900&display=swap" rel="stylesheet">

  <script type="text/javascript" src="https://d3js.org/d3.v4.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/d3-tip/0.9.1/d3-tip.min.js"></script>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/d3/7.8.5/d3.min.js"></script>
</head>

<body>
  <script src="functions.js"></script>
  <script src="constants.js"></script>
  <script src="utils.js"></script>
  <script src="psiChart_nonLinear.js"></script>
  <script src="comparison.js"></script>
  <script src="networkVideos.js"></script>


  <header>

    <header class="full-width-header">
      <div class="header-content">
        <h1>Inside an interpretable-by-design machine learning model: enabling RNA splicing rational design</h1>
        <!-- <p class="subtitle">Subtitle</p> -->
        <div class="author-info">
          <p>
            <span class="author" data-tooltip="M.A. and S.Z. contributed equally to this work.">Mateus Aragao</span>, 
            <span class="author" data-tooltip="M.A. and S.Z. contributed equally to this work.">Shiwen Zhu</span>, 
            <span class="author">Nhi Nguyen</span>, 
            <span class="author">Alejandro Garcia</span>, 
            <span class="author">Susan E. Liao</span>
          </p>

          <p class="affiliation">Department of Computer Science, New York University</p>
          <!-- <p>Published on <time datetime="2024-07-09">July 9, 2024</time></p> -->
        </div>
      </div>
    </header>

    <main class="content-wrapper">
      <article>

        <section>
          <h2>Introduction</h2>
          <p>
            RNA splicing, a key process in human genetics, has garnered considerable interest from the 
            machine learning community. Splicing involves removing regions from RNA transcripts and 
            joining the remaining segments together. Alternative splicing enables a single gene to 
            generate multiple output transcripts. <br><br>

            Understanding how specific RNA features dictate splicing outcomes remains a major challenge 
            in computational biology. Several convolutional neural networks have been developed to 
            predict splicing outcomes, but many of these networks are opaque and cannot explain the 
            reasoning that leads to their prediction. We previously introduced an "interpretable-by-design" 
            neural network to examine splicing logic. In contrast to these earlier models, our model is 
            both globally and locally interpretable, following a unified logic that can be used to trace 
            the decision-making process for each data point. <br><br>
            Here, we introduce a series of interactive visualizations that trace the model's reasoning 
            from input to output. This ability to visually dissect the model’s reasoning is particularly 
            valuable for the rational design of exons with targeted splicing results. It facilitates more 
            accurate manipulation of genetic sequences, allowing for the achievement of desired biological 
            effects with greater precision. Our goal with these visualizations is to provide a richer insight 
            into the mechanisms of splicing, thereby empowering researchers to predict and control splicing outcomes.
          </p>
        </section>
        <hr> <!-- horizontal line -->

        <h2>Background</h2>

        <h3><b>RNA Splicing</b></h3>
        <p>The most common form of alternative splicing is exon skipping, 
          which can lead to two distinct outcomes: exon inclusion or exon skipping. 
          Before splicing, RNA transcripts are composed of exons interrupted by introns. 
        </p>

        <!-- <h3><b>Exon Inclusion</b></h3> -->
        <p>
          In exon inclusion, all introns are removed and all exons are joined together in the final transcript.
        </p>
        <div class="container">
          <img id="previewImage" src="resources/RNA Splicing 2.001.jpeg" alt="Preview Image">
          <i class="fas fa-arrow-right"></i>
          <video id="myVideo" autoplay muted>
            <source src="resources/Splicing_Incl.m4v" type="video/mp4">
            Your browser does not support the video tag.
          </video>
          <button id="replayButton"><i class="fas fa-redo"></i></button>
        </div>

        <!-- <h3><b>Exon Skipping</b></h3> -->
        <p>
          Conversely, in exon skipping, an internal exon is removed and excluded from the final product.
        </p>
        <div class="container">
          <img id="previewImage" src="resources/RNA Splicing 2.001.jpeg" alt="Preview Image">
          <i class="fas fa-arrow-right"></i>
          <video id="myVideo2" width="480" height="270" autoplay muted>
            <source src="resources/Splicing_Skip.m4v" type="video/mp4">
            Your browser does not support the video tag.
          </video>
          <button id="replayButton2"><i class="fas fa-redo"></i></button>
        </div>

        <h3><b>Dataset for model training</b></h3>
        <p>
          Our interpretable model was trained on a splicing dataset of over 300,000 random exons, 
          all within a fixed sequence background. This controlled dataset structure enables the 
          neural network to learn features associated with exon skipping or inclusion in an unbiased manner. <br><br>
          Since single-stranded RNA can often adopt secondary structures, the network used two inputs for 
          each exon - the 70-nucleotide sequence and a predicted secondary structure. We represent 
          secondary structure using standard dot-parenthesis notation, where matching parantheses 
          indicate base-pairing and dots represent unpaired bases. 
        </p>

        <div class="ps-container">
            <div class="ps">
                <span class="inline-image">
                    <img src="resources/Cursors.002.jpeg" alt="Description of image">
                </span>
                Click on the dot-parantheneses notation to view the RNA structure.
            </div>
        </div>

        <!-- secondary structure -->
        <div class="container">
          <img id="previewImage3" src="resources/Secondary_structure.001.jpeg" alt="Preview" onclick="showVideo3()">
          <video id="myVideo3" autoplay muted>
            <source src="resources/Secondary_structure.m4v" type="video/mp4">
            Your browser does not support the video tag.
          </video>
          <button id="replayButton3"><i class="fas fa-redo"></i></button>
        </div>

        <p>
          <br><br>
          The dataset was fully-labelled, with each exon matched to experimentally measured 
          splicing outputs. The output is reported as percent spliced in (PSI) index, 
          the ratio of included exon products over all total post-splicing products. 
        </p>

        <div class="ps-container">
          <div class="ps">
              <span class="inline-image">
                  <img src="resources/Cursors.002.jpeg" alt="Description of image">
              </span>
              Click on Exon 1, Exon 2, or Exon 3 to explore different inputs and outputs.
          </div>
      </div>

        <!-- PSI Video Section -->
        <div class="video-section-container">
          <section id="video-section">
            <h3><b>Input</b></h3>
            <div class="video-buttons">
              <button class="styled-button one" onclick="selectVideo('video1', 'info1', 'one')">Exon 1</button>
              <button class="styled-button two" onclick="selectVideo('video2', 'info2', 'two')">Exon 2</button>
              <button class="styled-button three" onclick="selectVideo('video3', 'info3', 'three')">Exon 3</button>
            </div>
            <div id="info-container" class="info-container">
              <img src="" alt="Info Image" id="info-image">
            </div>
            <h3><b> Output</b></h3>
            <div class="video-container">
              <video id="video1" autoplay muted>
                <source src="resources/psiVideos/PSI_New_1.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
              <video id="video2" autoplay muted>
                <source src="resources/psiVideos/PSI_New_2.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
              <video id="video3" autoplay muted>
                <source src="resources/psiVideos/PSI_New_3.m4v" type="video/mp4"> Your browser does not support the
                video tag.
              </video>
            </div>
          </section>
        </div>
        <script src="psiVideos.js"></script>
        
        <hr> <!-- horizontal line -->
        
        <h2><b> Visualizing network interpretability</b></h2>
        
        <h3><b>Learning splicing features</b></h3>
        <p>
          The neural network was trained using fully-labeled data, where each exon input corresponds to a specific 
          PSI output. The network extracts information from the exon through one-hot encoding. In one-hot encoding, 
          sequence and structure information is converted into a binary vector. <br><br>
          For example, a nucleotide can be encoded as "hot" (1). The "hot" signal indicates whether the nucleotide 
          is A, C, G, or U. A matrix is used to represent a string of several nucleotides, where the nucleotide at 
          each position is hot-encoded.
        </p>
        <div class="encoding-video-container">
            <video id="videoA" muted autoplay>
                <source src="resources/Encoding_1.m4v" type="video/mp4">
                Your browser does not support the video tag.
            </video>
            <button id="replayButtonA"><i class="fas fa-redo"></i></button>
        </div>
        <p>
          One-hot encoding can also be used to represent RNA structural data. The "hot" signal can indicate 
          whether a position is unpaired (dot) or paired (parenthesis). 
        </p>
        <div class="encoding-image-container">
            <img src="resources/Encoding_Long.001 2.jpeg" alt="Description of the image" height="150px">
        </div>
      
        <p>
          During training, the network identifies features that lead to inclusion or skipping by 
          aggregating the one-hot encoded data. The frequency of a nucleotide or structure at a 
          given position in a matrix is visualized as a heat map - where darker red indicates 
          higher frequencies, and lighter colors represent lower frequencies.
        </p>
        <div class="encoding-video-container videoB-container">
          <video id="videoB" muted autoplay>
              <source src="resources/Encoding_2.m4v" type="video/mp4">
              Your browser does not support the video tag.
          </video>
          <button id="replayButtonB"><i class="fas fa-redo"></i></button>
      </div>
        <p>
          These heat maps can be visualized as sequence logos, where the height of each nucleotide 
          reflects its frequency at a specific position. In this example, the C nucleotides are 
          prominently favored at specific positions, whereas other nucleotides are less preferred. 
          Logo maps are commonly used in molecular biology to summarize conserved sequence and 
          structure patterns.
        </p>
        <div class="encoding-video-container videoC-container">
          <video id="videoC" muted autoplay>
              <source src="resources/Encoding_3.m4v" type="video/mp4">
              Your browser does not support the video tag.
          </video>
          <button id="replayButtonC"><i class="fas fa-redo"></i></button>
      </div>
        <p>
          After training, the network identified 20 total features that dictate splicing outcomes: 
          9 features contribute to exon inclusion, while 11 features contribute to exon skipping.
        </p>
        <div class="encoding-image-container">
          <img src="resources/Encoding_4.001.jpeg" alt="Description of the image" height="480px">
      </div>
        
        <!-- Network Video Section -->
        <h3><b>Examining how features inform splicing predictions</b></h3>
        <div class="network-video-wrapper">
        <div class="network-video-section-container" id="network-section-1">
            <section id="network-video-section-1">
              <div class="network-buttons">
                <div class="selector"></div>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-1', 'networkVideo1', 0)">Step 1<br>Input Exon</button>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-1', 'networkVideo2', 1)">Step 2<br>Learned Features</button>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-1', 'networkVideo3', 2)">Step 3<br>Class Strength</button>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-1', 'networkVideo4', 3)">Step 4<br>Difference-to-Prediction</button>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-1', 'networkVideo5', 4)">Step 5<br>Output Prediction</button>
              </div>
              <div class="network-video-container">
                <video id="networkVideo1" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Network_1_Updated.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                <video id="networkVideo2" muted preload="auto">
                  <source src="resources/networkVideos/Network_2_Updated.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                <button id="pauseButton1-2" class="pause-button hide-button" onclick="togglePause('networkVideo2', this)">||</button>
                <video id="networkVideo3" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Network_3_Updated.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                <video id="networkVideo4" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Network_4_Updated.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                <video id="networkVideo5" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Network_5_Updated.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                
              </div>
              <p id="network-description-1">The network uses two inputs for each exon - the 70-nucleotide sequence and a predicted secondary structure. </p>
            </section>
          </div>
        </div>

        <p>
          In the final stage of the network, the network uses a learned non-linear function to convert the difference 
          between inclusion and skipping strengths into a predicted PSI.
        </p>
        
        <div class="ps-container">
          <div class="ps">
              <span class="inline-image">
                  <img src="resources/Cursors.002.jpeg" alt="Description of image">
              </span>
              Click on points along the function to illustrate the non-linear relationship between Δ Strength and predicted PSI.
          </div>
      </div>
      <div class="chart-container">
        <div id="non-linear-chart" class="non-linear-chart"></div>
        <div id="psi-bar-chart" class="psi-bar-chart"></div>
      </div>

        <h3><b>Examining predictions for specific inputs</b></h3>
        <p>
          Using this framework, we can also examine the local interpretability of the network on an individual exon.
        </p>
        <div class="ps-container">
            <div class="ps">
                <span class="inline-image">
                    <img src="resources/Cursors.002.jpeg" alt="Description of image">
                </span>
                Click on each step of to explore network logic on a specific input exon.
            </div>
        </div>

        <!-- Local Network Video Section -->
        <div class="network-video-wrapper">
          <div class="network-video-section-container" id="network-section-2">
            <section id="network-video-section-2">
              <div class="network-buttons">
                <div class="selector"></div>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo1', 0)">Step 1<br>Input Exon</button>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo2', 1)">Step 2<br>Learned Features</button>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo3', 2)">Step 3<br>Class Strength</button>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo4', 3)">Step 4<br>Difference-to-Prediction</button>
                <button class="network-styled-button" onclick="selectNetworkVideo('network-section-2', 'newNetworkVideo5', 4)">Step 5<br>Output Prediction</button>
              </div>
              <div class="network-video-container">
                <video id="newNetworkVideo1" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Network_1_Updated.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                <video id="newNetworkVideo2" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Local_Network_Feature.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                <button id="pauseButton2-2" class="pause-button hide-button" onclick="togglePause('newNetworkVideo2', this)">||</button>
                <video id="newNetworkVideo3" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Local_Network_Add.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                <video id="newNetworkVideo4" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Local_Network_Subt.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
                <video id="newNetworkVideo5" autoplay muted preload="auto">
                  <source src="resources/networkVideos/Local_Network_PSI.m4v" type="video/mp4"> Your browser does not support the video tag.
                </video>
              </div>
              <p id="network-description-2">The network considers the input exon's sequence and structure.</p>
            </section>
          </div>
        </div>

        <p>
          We further introduce an interactive exon view that plots the strength contributions along the length of the exon. 
          These visualizations map the inclusion and skipping strengths at nucleotide resolution. By clicking on an individual 
          position in an exon, users highlight the sequence, structure, and features at the given position. We emphasize how a 
          single nucleotide can contribute to multiple features by highlighting the relevant features in the legend, sort 
          the features by strength in a hierarchical bar chart in nucleotide view, and provide a high-resolution map of 
          features in nucleotide features.
        </p>


        <div class="ps-container">
          <div class="ps">
              <span class="inline-image">
                  <img src="resources/Cursors.002.jpeg" alt="Description of image">
              </span>
              Click on an individual nucleotide along the exon to highlight specific features that map to that nucleotide.
          </div>
      </div>

        <div class="feature-nucleotide">
          <!-- <button id="switch-dataset-button">Switch Dataset</button> -->
          <button id="replayButtonNucleotideView"><i class="fas fa-redo"></i></button>

          <div class="nucleotide-view" id="nucleotide-view">
            <svg class="nucleotide-view" width="100%" height="100%"></svg>
          </div>
          <div class="bottom">
            <div class="feature-legend-container">
              <div class="feature-legend-grids">
                <!-- <div class="feature-legend-title">Feature Legend</div> -->
                <div class="legend"></div>
                <div class="feature-grids-wrapper">
                  <div class="svg-grid-inclusion"></div>
                  <div class="svg-grid-skipping"></div>
                </div>
                <div class="svg-grid-long-skipping"></div>
              </div>
            </div>
            <div class="nucleotide-mini-charts">
              <div class="nucleotide-sort">
                <svg class="nucleotide-sort" width="100%" height="100%"></svg>
              </div>
              <div class="nucleotide-zoom">
                <svg class="nucleotide-zoom" width="100%" height="100%"></svg>
              </div>
            </div>
          </div>
        </div>

        <hr>

        <h2> Rational design applications</h2>

        <p>One key aspect of interpretability is comparing the network’s reasoning process across multiple inputs. 
          At the local level, this involves examining how the network’s complex reasoning differs across input 
          instances, specifically examining how the network identifies different features across inputs yet follows 
          the same reasoning process to arrive at similar predictions. By analyzing reasoning across multiple input 
          instances, researchers can also gain insights into the global reasoning processing within a network. 
          The visualizations mapping inclusion and skipping features along the length of the exon demonstrate the 
          local interpretability of our model. These visualizations can be applied to a range of different questions 
          regarding rational design. Here, we focus on three specific applications: <br><br>
          (1) Comparing different exon inputs with the same output prediction <br>
          (2) Examining effects of mutations on splicing outcomes, and <br>
          (3) Generalizing to exons from other datasets.
        </p>

        <h3><b>Comparing different exon inputs with the same output prediction </b></h3>

        <p>
          One key aspect of interpretability is examining the network’s reasoning across different inputs. 
          At the local level, this involves examining how the network’s complex reasoning differs across input instances,
           specifically examining how the network identifies different features across inputs yet follows the same reasoning 
           process to arrive at similar predictions. 
        </p>
        <div class="chart-container">
          <svg id="nucleotide-view-exon1" class="nucleotide-view-exon1"></svg>
      </div>
      <br>
      <div class="chart-container">
          <svg id="nucleotide-view-exon-comp" class="nucleotide-view-exon-comp"></svg>
      </div>

        <p>
          Our first use case for rational design demonstrates how different exons can result in the same predicted PSI. 
          We examined two exons from our dataset that had dramatically different inclusion and skipping profiles. 
          While the profiles of these exons were different, the calculated Δ Strength of both exons was -13.9, 
          leading to a predicted PSI output of 0.15.
<br><br>
          We can further extend our analysis to observe multiple exons with the same PSI output.
        </p>

        <!-- 3x3 grid of graphs -->
        <div class="grid-container">
          <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-1" class="nucleotide-view-exon-grid"></svg>
          </div>
          <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-2" class="nucleotide-view-exon-grid"></svg>
          </div>
          <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-3" class="nucleotide-view-exon-grid"></svg>
          </div>
          <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-4" class="nucleotide-view-exon-grid"></svg>
          </div>
          <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-5" class="nucleotide-view-exon-grid"></svg>
          </div>
          <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-6" class="nucleotide-view-exon-grid"></svg>
          </div>
          <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-7" class="nucleotide-view-exon-grid"></svg>
          </div>
          <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-8" class="nucleotide-view-exon-grid"></svg>
          </div>
          <!-- <div class="grid-item">
            <svg id="nucleotide-view-exon-s1-9" class="nucleotide-view-exon-grid"></svg>
          </div> -->
        </div>
        <h3><b>Examining effects of mutations on splicing outcomes </b></h3>
        <p>
          A common benchmark for the performance of computational models is to experimentally 
          validate predictions. One common approach is to perform single nucleotide mutagenesis 
          - substituting one nucleotide for another (e.g. replacing A with G at a specific position 
          in an exon). For machine learning researchers, mutagenesis provides an approach to 
          probe whether the local interpretability of a neural network generalizes beyond the 
          training and testing datasets. For molecular biology experts, comparing how well predictions
          from a computational model align with experimental laboratory measurements is a key indicator 
          of model trustworthiness and interpretability. Furthermore, 
          mutagenesis experiments are often used to model real-world disease mutations as single 
          nucleotide splicing mutations have been directly implicated in several diseases. To 
          illustrate how our visualizations can be useful for mutagenesis modeling, we generate 
          exon view visualizations for a single mutation. Each visualization elucidates important 
          facets of the network's local reasoning process. 
        </p>
                <div class="ps-container">
          <div class="ps">
              <span class="inline-image">
                  <img src="resources/Cursors.002.jpeg" alt="Description of image">
              </span>
              Click on the exon labels to explore how a single mutation changes inclusion and skipping strengths.
          </div>
      </div>
        <div class="chart-container">
          <svg id="nucleotide-view-exon-s1-mutation" class="nucleotide-view-exon-s1-mutation"></svg>
        </div>

        <h3><b>Generalizing to exons with disease relevance</b></h3>
        <p>
          Finally, while our model was trained on a dataset of random exons with a fixed length, 
          we previously demonstrated that after minimal adjustments, our model accurately predicts 
          PSI for exons from the human genome. This demonstrates that our model generalizes to 
          other exons outside of the training dataset, highlighting the trustworthiness and 
          interpretability of the model.
        <br><br>
        The interpretability of our model features also has important implications for biomedical research. 
        Here, we highlight how our visualizations can demonstrate the divergent impact of two mutations on BRCA2, 
        one of the strongest genetic predictors of breast and ovarian cancer. In healthy adults, BRCA2 exon 7 is 
        entirely included. However, the Human Gene Mutation Database reveals that many breast and ovarian cancer 
        patients have mutations in BRCA2 exon 7. Thus, understanding whether specific BRCA2 exon 7 mutations 
        impact splicing outcomes can  provide valuable insights into molecular disease mechanisms, identify 
        candidate biomarkers to trace disease progression, and guide the development of targeted therapies.
        <br><br>
        Some mutations have minimal effects on BRCA2 exon 7 splicing. For example, the 539T>C mutation 
        disrupts few splicing features and does not significantly alter PSI.
        </p>

      <!-- <p>BRCA2</p> -->
      <div class="chart-container">
        <svg id="nucleotide-view-exon-brca2" class="nucleotide-view-exon-brca2"></svg>
      </div>

      <p>
        In contrast, the 620C>T mutation introduces multiple changes in predicted BRCA2 exon 7 structure. 
        This single mutation dramatically changes the exon view silhouette to favor exon skipping.
      </p>

      <div class="chart-container">
        <svg id="nucleotide-view-exon-brca2-3" class="nucleotide-view-exon-brca2-3"></svg>
      </div>

      <p></p>
      These examples highlight how interpretable visualization can guide researchers in determining 
      the specific features that mutations affect.

      </p>

      <hr> <!-- horizontal line -->
<!--         
      <h3><b> User-designed exons</b></h3>
      <p>to-do</p> -->
        <h2>Conclusion</h2>
        <p>
          In this interactive article, we present a series of visualizations designed to 
          enhance our understanding of an interpretable machine learning model. These novel 
          visualizations offer a detailed and intuitive way to map the features learned by the 
          model onto specific input sequences. By providing a clear visual representation of how 
          the model interprets and processes different splicing events, these tools allow researchers 
          to see exactly how various features dictate splicing outcomes. This capability is 
          particularly valuable for designing and engineering specific exons with targeted 
          splicing results, facilitating more precise manipulation of genetic sequences to achieve 
          desired biological effects. Through these visualizations, we aim to provide a deeper 
          insight into the mechanisms of splicing and improve our ability to predict and control 
          splicing outcomes in genetic research and applications.
          <br><br>
          <br><br>
        </p>
        <!-- <div class="collapsible-section">
          <button class="collapsible button">Learn More(example of a foldable section)</button>
          <div class="content">
            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et
              dolore
              magna aliqua.</p>
          </div>
        </div> -->


      </article>
    </main>
    <footer class="article-footer">
      <div class="footer-content">
        <section>
          <h3>Acknowledgment</h3>
          <p>
            We thank Oded Regev, Simon Liu, Elizabeth Wei， Allison Shi, Nebiyu Ayele (NYU), 
            Gloria M. Sheynkman (UVA), and Jingyi Fei (UChicago) for thoughtful comments 
            and suggestions on visualizations. SZ is supported by the NYU Dean's Undergraduate 
            Research Fund. AG was supported by the NYU Center for Data Science Undergraduate 
            Research Program. SEL is supported by an NSF Collaborative Research Grant on Innovative 
            Machine Learning Methods for Research on Biomacromolecular Systems (2226731) and an 
            Additional Ventures Career Development Award. 
          </p>
        </section>

        <!-- <section>
          <h3>Discussion and Review</h3>
          <p>if there are any discussions</p>
        </section> -->

        <section>
          <h3>Further Reading</h3>
          <ul>
            <li><a href="https://interpretable-machine-learning-interface-demo.vercel.app" style=" text-decoration: underline;">Interpretable machine learning interface</a></li>
            <li><a href="https://www.pnas.org/doi/10.1073/pnas.2221165120" style="text-decoration: underline;">Deciphering RNA splicing logic with interpretable machine learning</a></li>
          </ul>
        </section>
      </div>
    </footer>
</body>

</html>